module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/MakeLinkTest.php

```diff
@@ -0,0 +1,219 @@
+        $view = $this->createMock(\Laminas\View\Renderer\PhpRenderer::class);
+        $view->method('plugin')->will($this->returnValue($escapeHtml));
+
+        $helper = new MakeLink();
+        $helper->setView($view);
+        return $helper;
+    }
+
+    /**
+     * Test that responds to common inputs
+     *
+     * @return void
+     */
+    public function testLink()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<a href="https://vufind.org">text</a>',
+            $helper('text', 'https://vufind.org')
+        );
+
+        $this->assertEquals(
+            '<a href="/Record/id">text</a>',
+            $helper('text', '/Record/id')
+        );
+
+        $this->assertEquals(
+            '<a href="#anchor">text</a>',
+            $helper('text', '#anchor')
+        );
+
+        $this->assertEquals(
+            '<a href="#">text</a>',
+            $helper('text', '#')
+        );
+    }
+
+    /**
+     * Test that helper returns unescaped text when href is falsey
+     *
+     * @return void
+     */
+    public function testNoLink()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals('text', $helper('text', ''));
+        // Escapes
+        $this->assertEquals('text&', $helper('text&', null));
+        $this->assertEquals('text<', $helper('text<', false));
+    }
+
+    /**
+     * Test that responds to common inputs
+     *
+     * @return void
+     */
+    public function testAttributes()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<a class="btn" id="login" href="#">text</a>',
+            $helper('text', '#', ['class' => 'btn', 'id' => 'login'])
+        );
+
+        // Skip href
+        $this->assertEquals(
+            '<a href="#" class="btn" id="login">text</a>',
+            $helper('text', ['href' => '#', 'class' => 'btn', 'id' => 'login'])
+        );
+
+        // String
+        $this->assertEquals(
+            '<a class="btn" href="#">text</a>',
+            $helper('text', '#', 'btn')
+        );
+
+        // No href but attributes
+        $this->assertEquals(
+            '<span class="btn">text</span>',
+            $helper('text', null, 'btn')
+        );
+        $this->assertEquals(
+            '<span class="btn">text</span>',
+            $helper('text', ['class' => 'btn'])
+        );
+        $this->assertEquals(
+            '<span class="btn">text</span>',
+            $helper('text', null, ['class' => 'btn'])
+        );
+    }
+
+    /**
+     * Test escaping with other plugins
+     *
+     * @return void
+     */
+    public function testEscaping()
+    {
+        $helper = $this->getHelper();
+
+        $recordLink = $this->getRecordLink();
+        $this->assertEquals(
+            '<a href="/Record/foo">recordLink</a>',
+            $helper('recordLink', $recordLink->getUrl('Solr|foo'))
+        );
+
+        $urlHelper = $this->getUrl();
```

Comment:
$urlHelper doesn't seem to get used at all -- can we get rid of this and the getUrl method entirely?

Reply:
@crhallberg, I see that you removed one reference to getUrl(), but you're still creating a URL helper in the helper container you're using in your test, but as far as I can see, the MakeLink helper never uses it for anything. Am I missing something?

Reply:
The test seems to break when I don't include it because RecordLink seems to want it and I want to actually use RecordLink for the test unless you are strongly against it.

Reply:
Well, my argument is that what we are testing here is that MakeLink deals correctly with the output of RecordLink. We have a different existing RecordLink test that tests whether RecordLink returns correct output when given specific input. I don't think we need to repeat that test as part of MakeLinkTest -- we can just short-circuit it and provide the known output of RecordLink here and test that MakeLink handles it correctly. Since this is a unit test, it's only supposed to be testing one piece of code, and incorporating RecordLink makes it more of an integration test. Maybe I'm just being overly strict about definitions, but my main objection is that we're already doing this exact same work in a different test, so it seems like a waste of effort to do it again here when it's kind of "out of scope" anyway.